src/shared/serverinfo.hpp

```diff
@@ -0,0 +1,231 @@
+    // TODO: remove it. Used by:
+    //   _ resolverquery() in engine/serverbrowser.cpp
+    //   _ servercompare() in game/client.cpp
+    char const* name( void ) const;
+
+    // "return" a property of this server to CS
+    // if property < 0: returns the number of properties one can query (4)
+    // if property > 3: do nothing.
+    // if property > 0:
+    //   if index < 0: returns the size of requested array
+    //   if index > array.size(): do nothing
+    //   if property == 1: returns m_attr[index] (integer)
+    //   if property == 2: returns players[index] (result?)
+    //   if property == 3: returns handles[index] (result?)
+    // if propert == 1:
+    //   if index ==  0: returns server_status() (integer)
+    //   if index ==  1: returns name (result) (this is probably the IP or domain name)
+    //   if index ==  2: returns port (integer)
+    //   if index ==  3: returns description or name if no description (result)
+    //   if index ==  4: returns map played (result)
+    //   if index ==  5: returns number of players (integer)
+    //   if index ==  6: returns ping? Guess it's roundtrip time? (integer)
+    //   if index ==  7: returns m_lastinfo? guess it's time of last update? (integer)
+    //   if index ==  8: returns authhandle?? (result)
+    //   if index ==  9: returns flags??  (result)
+    //   if index == 10: returns branch?? (resull)
+    //   if index == 11: returns priority (integer)
+    void cube_get_property( int property, int index );
+
+    // write some current server's attributes into file
+    // * m_name
+    // * port
+    // * priority
+    // * description() (escaped)
+    // * authhandle (escaped)
+    // * flags (escaped)
+    // * branch (escaped)
+    void writecfg( stream& file ) const;
+
+    // update server's internal informations.
+    // According to what exactly is still unknown to me.
+    // Candidate for rewrite.
+    // TODO: make private?
+    //   * only used in serverbrowser.cpp
+    void update( size_t len, void const* data, int serverdecay, int totalmillis, int lastreset );
+
+    // methods that allows multi-threaded server name resolution without breaking encapsulation
+    bool validate_resolve( char const* name, ENetAddress const& addr );
+    bool need_resolve( int& resolving );
+
+    // check if current's server network info (IP/port) match parameter(s)
+    bool is_same( ENetAddress const& addr ) const;
+    bool is_same( char const* oname, int oport ) const;
+
+    // sends a ping to target
+    void ping( ENetSocket& sock, int serverdecay, int millis );
+
+    // register a new server into "servers"
+    static serverinfo *newserver(const char *name, int port = SERVER_PORT, int priority = 0, const char *desc = nullptr, const char *handle = nullptr, const char *flags = nullptr, const char *branch = nullptr, uint ip = ENET_HOST_ANY);
+
+    // check if the target is compatible with current client
+    static bool server_compatible( serverinfo const* target );
```

Comment:
Naming wise, this should likely be prefixed `is_`. The motivation doesn't get clear from the docstring, though. Could you please explain what "compatible" means?